### PR TITLE
Introduced parent and parents keywords for binding.

### PR DESCRIPTION
**Pull request by nsndeck**

### CODE REVIEW

**Comment by enchev** [comment]
:+1:

File: ui/builder/binding-builder.ts [hunk 1]
```diff
@@ -87,9 +90,30 @@ function extractPropertyNameFromExpression(expression: string): string {
     }
 }
 
+function getParamsArray(value: string) {
```
Comment (hshristov) [inline ui/builder/binding-builder.ts]: You could specify the return value of the function.

File: ui/core/bindable.ts [hunk 2]
```diff
@@ -197,6 +200,14 @@ export class Binding {
         }
     }
 
+    private prepareExpressionForUpdate(): string {
+        var escapeRegex = /[-\/\\^$*+?.()|[\]{}]/g;
```
Comment (hshristov) [inline ui/core/bindable.ts]: Please add comment what this regex does.

File: ui/core/bindable.ts [hunk 3]
```diff
@@ -21,6 +21,9 @@ function onBindingContextChanged(data: dependencyObservable.PropertyChangeData)
 }
 
 var contextKey = "context";
+var paramsRegex = /\[\s*(['"])*(\w*)\1\s*\]/;
```
Comment (hshristov) [inline ui/core/bindable.ts]: Please add comments what these regexes do.

**Comment by hshristov** [comment]
Added few comments. :+1:

**Pull request merged**